code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthTokenDaoImpl.java

```diff
@@ -77,4 +80,16 @@ public AuthToken getTokenByAccountId(long accountId) {
     public boolean deleteToken(String key) {
         return create().executeDelete(getTokenByKey(key)) == 1;
     }
+
+    @Override
+    public List<AuthToken> getTokenByAuthenticatedAsAccountId(long authenticatedAsAccountId, long tokenAccountId) {
+        return create()
+                .selectFrom(AUTH_TOKEN)
+                .where(AUTH_TOKEN.AUTHENTICATED_AS_ACCOUNT_ID.eq(authenticatedAsAccountId))
+                        .and(AUTH_TOKEN.ACCOUNT_ID.eq(tokenAccountId))
+                        .and(AUTH_TOKEN.VERSION.eq(SecurityConstants.TOKEN_VERSION))
+                        .and(AUTH_TOKEN.PROVIDER.eq(SecurityConstants.AUTH_PROVIDER.get()))
+                        .and(AUTH_TOKEN.EXPIRES.greaterThan(new Date()))
+                .orderBy(AUTH_TOKEN.EXPIRES.desc()).fetch().into(AuthToken.class);
```

Comment:
What is the point of the 
```
.and(AUTH_TOKEN.VERSION.eq(SecurityConstants.TOKEN_VERSION))
.and(AUTH_TOKEN.PROVIDER.eq(SecurityConstants.AUTH_PROVIDER.get()))
``` 
conditions? Seems like the only thing we should care about is the two ids.
Checking the expires seems fine to me, though also not entirely necessary.